doc/bootstrap-a-new-developer-macbook.md

```diff
@@ -0,0 +1,41 @@
+# Bootstrapping a new developer's Macbook
+
+# Ensure you have admin rights
```

Comment:
Running `sudo -v` would prompt a password, right? Wouldn't that affect the remaining shell session and how all the remaining bootstrapping is installed? Or does that only occur when prefixing each install command with sudo i.e. `sudo brew install ....`?

I'm don't have a lot of bash experience here, so correct me if I'm wrong, but I've heard it said before that needlessly installing thing with _sudo_ would come to haunt you down the line. So when you're installing other things later which might depend on your previous installs that are now _sudo_ installed, these new will also need sudo

Reply:
Great that you ask 😉 but no, it doesn't elevate the permissions for the rest of the commands. What it does is that it asks for the password upfront, and then the following `sudo foo` commands will not get stuck asking for the password, given that the commands run within a certain time period as the manpage says. That's the only read change.

From `man sudo`:

```
       -v, --validate
                   Update the user's cached credentials, authenticating the user
                   if necessary.  For the sudoers plugin, this extends the sudo
                   timeout for another 5 minutes by default, but does not run a
                   command.  Not all security policies support cached
                   credentials.
```

PS: this is a great way of doing an autopsy on a computer, because you can immediately type `sudo -v` and then ask the owner to type in his password (if they trust you :) ). From there on, if needed, you can run `sudo` commands without running around after the owner, or even worse calling them and asking for them to tell you the password 🤦‍♂️